parse.py

```diff
@@ -14,25 +20,18 @@
 import mwparserfromhell
 import nltk
 
-from keras.models import load_model
-from keras.preprocessing.sequence import pad_sequences
-from keras import backend as K
-K.clear_session()
-K.set_session(K.tf.Session(config=K.tf.ConfigProto(intra_op_parallelism_threads=10, inter_op_parallelism_threads=10)))
-
 cfg = config.get_localized_config()
 WIKIPEDIA_BASE_URL = 'https://' + cfg.wikipedia_domain
 
-def load_citation_needed():
-    # load the vocabulary and the section title dictionary
-    vocab_w2v = pickle.load(open(cfg.vocb_path, 'rb'))
-    section_dict = pickle.load(open(cfg.section_path, 'rb'), encoding='latin1')
+self = types.SimpleNamespace() # Per-process state
 
-    # load Citation Needed model
-    model = load_model(cfg.model_path)
-    return model, vocab_w2v, section_dict
+def initializer():
+    from keras.preprocessing.sequence import pad_sequences
+    self.vocab_w2v = pickle.load(open(cfg.vocb_path, 'rb'))
+    self.section_dict = pickle.load(open(cfg.section_path, 'rb'), encoding='latin1')
+    self.pad_sequences = pad_sequences
```

Comment:
At this point we might just want to move this stuff to KerasModel and also move the logic in `run_citation_needed` to `KerasModel.predict`? OK if you want to leave it to a follow-up PR though.

(As it is in this PR, it's a bit odd that `run_citation_needed` doesn't actually run the model despite its name.)

Reply:
Yeah, it looks better to move all model-related stuff to keras_model !